### PR TITLE
More intelligent drop / create for tables

### PR DESCRIPTION
**Pull request by zachmu**
This addresses #2390 and other issues.

Previously we were only reusing dropped tables if they had exactly the same schema upon recreation. This PR loosens that constraint.

### CODE REVIEW

File: integration-tests/bats/drop-create.bats [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+    dolt sql -q "drop table test"
+
+    run dolt sql -q "show tables"
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ "test" ]] || false
+
+    dolt sql --disable-batch <<SQL
+create table test(a bigint primary key, b tinyint null, c int null);
+insert into test(a,b) values (1,1), (2,2);
```
Comment (timsehn) [inline integration-tests/bats/drop-create.bats:238]: Maybe a few other alters like adding defaults, constraints or triggers.

**Review by timsehn — COMMENTED** [review]
A bit more test coverage

File: integration-tests/bats/drop-create.bats (hunk 1 above)
Comment (zachmu) [reply integration-tests/bats/drop-create.bats:243]: Take another look

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM

**Pull request merged**